LoopTests/Mock Stores/MockCarbStore.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  Created by Anna Quinlan on 8/7/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import HealthKit
+import LoopKit
+@testable import Loop
+
+class MockCarbStore: CarbStoreProtocol {
+    init(for test: DataManagerTestType = .flatAndStable) {
+        self.testType = test // The store returns different effect values based on the test type
+    }
+    
+    var testType: DataManagerTestType
+    
+    var sampleType: HKSampleType = HKSampleType.quantityType(forIdentifier: HKQuantityTypeIdentifier.dietaryCarbohydrates)!
+    
+    var preferredUnit: HKUnit! = .gram()
+    
+    var delegate: CarbStoreDelegate?
+    
+    var carbRatioSchedule: CarbRatioSchedule?
+    
+    var insulinSensitivitySchedule: InsulinSensitivitySchedule?
+    
+    var insulinSensitivityScheduleApplyingOverrideHistory: InsulinSensitivitySchedule? = InsulinSensitivitySchedule(
+        unit: HKUnit.milligramsPerDeciliter,
+        dailyItems: [
+            RepeatingScheduleValue(startTime: 0.0, value: 45.0),
+            RepeatingScheduleValue(startTime: 32400.0, value: 55.0)
+        ],
+        timeZone: .utcTimeZone
+    )!
+    
+    var carbRatioScheduleApplyingOverrideHistory: CarbRatioSchedule?
+    
+    var maximumAbsorptionTimeInterval: TimeInterval {
+        return defaultAbsorptionTimes.slow * 2
+    }
+    
+    var delta: TimeInterval = .minutes(5)
+    
+    var defaultAbsorptionTimes: CarbStore.DefaultAbsorptionTimes = (fast: .minutes(30), medium: .hours(3), slow: .hours(5))
+    
+    var authorizationRequired: Bool = false
+    
+    var sharingDenied: Bool = false
+    
+    var healthStore: HKHealthStore = HKHealthStoreMock()
+    
+    func authorize(toShare: Bool, _ completion: @escaping (HealthKitSampleStoreResult<Bool>) -> Void) {
+        completion(.success(true))
+    }
+    
+    func replaceCarbEntry(_ oldEntry: StoredCarbEntry, withEntry newEntry: NewCarbEntry, completion: @escaping (CarbStoreResult<StoredCarbEntry>) -> Void) {
+        completion(.failure(.notConfigured))
+    }
+    
+    func addCarbEntry(_ entry: NewCarbEntry, completion: @escaping (CarbStoreResult<StoredCarbEntry>) -> Void) {
+        completion(.failure(.notConfigured))
+    }
+    
+    func getCarbStatus(start: Date, end: Date?, effectVelocities: [GlucoseEffectVelocity]?, completion: @escaping (CarbStoreResult<[CarbStatus<StoredCarbEntry>]>) -> Void) {
+        completion(.failure(.notConfigured))
+    }
+    
+    func generateDiagnosticReport(_ completion: @escaping (String) -> Void) {
+        completion("")
+    }
+    
+    func glucoseEffects<Sample>(of samples: [Sample], startingAt start: Date, endingAt end: Date?, effectVelocities: [GlucoseEffectVelocity]?) throws -> [GlucoseEffect] where Sample : CarbEntry {
+        return []
+    }
+    
+    func getCarbsOnBoardValues(start: Date, end: Date?, effectVelocities: [GlucoseEffectVelocity]?, completion: @escaping ([CarbValue]) -> Void) {
+        completion([])
+    }
+    
```

Comment:
I didn't have these functions to return any meaningful values because they aren't used in the portion of the algo that I'm testing